### PR TITLE
Replace destats with stats

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: src/dstats.jl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mutable struct DDEStats
+mutable struct DStats
```
Comment (devmotion) [inline src/dstats.jl:1]: Can you explain the reasoning for renaming the struct? It's internal anyway, so the name should not matter?

Comment (devmotion) [reply src/dstats.jl:1]: And the name `DDEStats` seems quite fitting.

Comment (devmotion) [reply src/dstats.jl:1]: Ah maybe you just did a quick search/replace `destats` -> `stats`? In that case, can you revert these undesired changes?

**Comment by avik-pal** [comment]
Locally tests pass with https://github.com/SciML/OrdinaryDiffEq.jl/pull/1889, https://github.com/SciML/DiffEqBase.jl/pull/886, and https://github.com/SciML/SciMLBase.jl/pull/391

File: Project.toml [hunk 2]
```diff
@@ -24,7 +24,7 @@ DiffEqBase = "6"
 OrdinaryDiffEq = "6.33"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "1.89"
```
Comment (ChrisRackauckas) [inline Project.toml:27]:
```suggestion
SciMLBase = "1.90"
```

**Pull request merged**